C front-end: tolerate type differences with asm renaming

tautschnig opened the pull request:
The example found in glibc shows how asm renaming may be combined with changes in return type names (even when there isn't actually a bit-level difference amongst the types). Weakening the asm-renamed declaration by marking it "incomplete" works around this problem.

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.

If your PR fixes a bug, include the regression test(s) in the same commit as the bug fix. Else, keep commits small and orthogonal, possibly placing tests in commits of their own.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

tautschnig committed:
C front-end: tolerate type differences with asm renaming

The example found in glibc shows how asm renaming may be combined with
changes in return type names (even when there isn't actually a bit-level
difference amongst the types). Weakening the asm-renamed declaration by
marking it "incomplete" works around this problem.

peterschrammel commented on src/ansi-c/c_typecheck_base.cpp:
```diff
@@ -646,6 +646,8 @@ void c_typecheck_baset::apply_asm_label(
   {
     symbol.name=asm_label;
     symbol.base_name=asm_label;
+    if(symbol.type.id() == ID_code)
```
An explaining comment would be great here.

tautschnig committed:
Implement all *scanf regression tests

We already had working tests for scanf and fscanf, and can easily extend
this to all variants.
